File: JavaScript/Math-Array-String/57.js
// 2574. Left and Right Sum Differences


/**
 * @param {number[]} nums
 * @return {number[]}
 */
var leftRightDifference = function(nums) {
    const n = nums.length;
    const leftSum = new Array(n).fill(0);
    const rightSum = new Array(n).fill(0);
    const answer = new Array(n).fill(0);

    // Calculate leftSum
    for (let i = 1; i < n; i++) {
        leftSum[i] = leftSum[i - 1] + nums[i - 1];
    }

    // Calculate rightSum
    for (let i = n - 2; i >= 0; i--) {
        rightSum[i] = rightSum[i + 1] + nums[i + 1];
    }

    // Calculate answer
    for (let i = 0; i < n; i++) {
        answer[i] = Math.abs(leftSum[i] - rightSum[i]);
    }

    return answer;
};

// Example usage
const nums1 = [10, 4, 8, 3];
console.log(leftRightDifference(nums1)); // Output: [15, 1, 11, 22]

const nums2 = [1];
console.log(leftRightDifference(nums2)); // Output: [0]

File: Python/Math-Array-String/57.py
# 2574. Left and Right Sum Differences

from typing import List

class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        n = len(nums)
        leftSum = [0] * n
        rightSum = [0] * n
        answer = [0] * n

        # Calculate leftSum
        for i in range(1, n):
            leftSum[i] = leftSum[i - 1] + nums[i - 1]

        # Calculate rightSum
        for i in range(n - 2, -1, -1):
            rightSum[i] = rightSum[i + 1] + nums[i + 1]

        # Calculate answer
        for i in range(n):
            answer[i] = abs(leftSum[i] - rightSum[i])

        return answer
    

sol = Solution()
output = sol.leftRightDifference([10, 4, 8, 3])
print(output) # # Output: [15, 1, 11, 22]